### PR TITLE
Sign up

### PR DESCRIPTION
**Pull request by StachuDev**
Rejestracja i logowanie użytkownika

### CODE REVIEW

File: Client/src/components/modal/_modal.scss [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  // padding: 40px;
+  top: 15px;
+  left: 10%;
+  width: 60%;
+  z-index: 100;
+  overflow: hidden;
+
+  .modal__header {
+    background: #4f005f;
+    // display: flex;
```
Comment (MaksymAndriienko) [inline Client/src/components/modal/_modal.scss]: Usuwany komentarze

File: Client/src/pages/signIn/SignIn.tsx [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    reValidateMode: "onBlur",
+    defaultValues: {
+      email: "",
+      password: "",
+    },
+    resolver: zodResolver(loginSchema),
+  });
+
+  const onSubmit: SubmitHandler<loginTypes> = async (data) => {
+    console.log(data);
```
Comment (MaksymAndriienko) [inline Client/src/pages/signIn/SignIn.tsx]: Usuwamy console log

File: Client/src/pages/signUp/components/RegistrationAgreement.tsx [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+        </p>
+        <p className="wrapper__text--small">
+          Klikając "Zgadzam się" wyrażasz Zgodę na Subskrybcję. Aby ją zapisać i
+          wydrukowac twoją Zgodę subskrypcji proszę użyj twojej przeglądarki
+        </p>
+        <button
+          className="form-button button-primary"
+          name="agreement"
+          onClick={() => {
+            setPage!((prev) => prev + 1);
```
Comment (MaksymAndriienko) [inline Client/src/pages/signUp/components/RegistrationAgreement.tsx:39]: co to za !?

File: Client/src/pages/signUp/components/RegistrationForm.tsx [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+import { FormInput } from "../../../types/registrationTypes";
+import {
+  paymentsTypes,
+  displayArray,
+  UserTypes,
+  modalTypes,
+  profileTypes,
+} from "../../../types/registrationTypes";
+
+const display: displayArray = {
```
Comment (MaksymAndriienko) [inline Client/src/pages/signUp/components/RegistrationForm.tsx]: To nie jest array

File: Client/src/pages/signUp/components/RegistrationForm.tsx [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+
+        console.log("id użytkownika: ", userResponse.user.id);
+        const paymentsDataIndex = formData ? 1 : 0;
+        const paymenstResponse = await postPayment(
+          "payments",
+          paymentsData[paymentsDataIndex]
+        );
+
+        const profileResponse = await postProfile(`profiles`, profileData);
+        if (paymenstResponse && profileResponse) {
```
Comment (MaksymAndriienko) [inline Client/src/pages/signUp/components/RegistrationForm.tsx]: Nie jest potrzebny, całkiem usuwamy console

File: Client/src/pages/signUp/components/payments/PaymentsContent.tsx [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+          okresu rozliczeniowego. Nie zwracamy środków ani nie udzielamy
+          rekompensaty za częściowo wykorzystane miesiące lub lata.
+        </p>
+
+        <button type="submit" className="form-button button-primary">
+          Zgadzam się
+        </button>
+      </div>
+    );
+  } else {
```
Comment (MaksymAndriienko) [inline Client/src/pages/signUp/components/payments/PaymentsContent.tsx]: Nie potrzebny else

File: Client/src/pages/signUp/components/payments/RegistrationPayments.tsx [hunk 7]
```diff
@@ -0,0 +1,73 @@
+// import { useState } from "react";
```
Comment (MaksymAndriienko) [inline Client/src/pages/signUp/components/payments/RegistrationPayments.tsx]: Zakomentowane też usuwamy

File: Client/src/pages/signUp/components/payments/_payments-all.scss [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+    flex-direction: row;
+    gap: 20px;
+
+    .input {
+      width: 190px;
+      // display: inline-block;
+    }
+  }
+
+  .content__main-text {
```
Comment (MaksymAndriienko) [inline Client/src/pages/signUp/components/payments/_payments-all.scss:26]: NIe wykożystujesz scss

File: Client/src/style/fonts/roboto/LICENSE.txt [hunk 9]
```diff
@@ -0,0 +1,202 @@
+
```
Comment (MaksymAndriienko) [inline Client/src/style/fonts/roboto/LICENSE.txt]: Przez google fonts ten font można zaimportować

File: Client/src/utils/Posts.ts [hunk 10]
```diff
@@ -0,0 +1,64 @@
...
+  try {
+    const response = await axios.post(authURL + endpoint, data);
+    localStorage.setItem("jwt", response.data.jwt);
+    setAuthToken(response.data.jwt);
+    return response.data as UserPostResponseTypes;
+  } catch (err) {
+    if (isAxiosError(err)) {
+      if (err.response?.status === 400) {
+        throw new Error("Użytkownik o takim emailu już istnieje.\nSpróbuj ponownie!");
+      }else{
```
Comment (MaksymAndriienko) [inline Client/src/utils/Posts.ts:41]: else nie jest potrzebny

File: Client/src/components/modal/_modal.scss (hunk 1 above)
Comment (StachuDev) [reply Client/src/components/modal/_modal.scss]: commit: [removing console.log and comment out code](https://github.com/MaciejMakowskiBille/NetflixClone/commit/1e9d40099cb4e4b904eaf00df4e823c9feb9466d)

File: Client/src/pages/signIn/SignIn.tsx (hunk 2 above)
Comment (StachuDev) [reply Client/src/pages/signIn/SignIn.tsx]: commit: [removing console.log and comment out code](https://github.com/MaciejMakowskiBille/NetflixClone/commit/1e9d40099cb4e4b904eaf00df4e823c9feb9466d)

File: Client/src/pages/signUp/components/RegistrationForm.tsx (hunk 4 above)
Comment (StachuDev) [reply Client/src/pages/signUp/components/RegistrationForm.tsx]: commit: [tidy up code, pageArray type name change](https://github.com/MaciejMakowskiBille/NetflixClone/commit/7f820804210f6dd4d5f4568d0d2fead3afae1a44)

File: Client/src/pages/signUp/components/RegistrationForm.tsx (hunk 5 above)
Comment (StachuDev) [reply Client/src/pages/signUp/components/RegistrationForm.tsx]: commit: [tidy up code, pageArray type name change](https://github.com/MaciejMakowskiBille/NetflixClone/commit/7f820804210f6dd4d5f4568d0d2fead3afae1a44)

File: Client/src/pages/signUp/components/payments/PaymentsContent.tsx (hunk 6 above)
Comment (StachuDev) [reply Client/src/pages/signUp/components/payments/PaymentsContent.tsx]: commit: [tidy up code, pageArray type name change](https://github.com/MaciejMakowskiBille/NetflixClone/commit/7f820804210f6dd4d5f4568d0d2fead3afae1a44)

File: Client/src/pages/signUp/components/payments/RegistrationPayments.tsx (hunk 7 above)
Comment (StachuDev) [reply Client/src/pages/signUp/components/payments/RegistrationPayments.tsx]: commit: [removing console.log and comment out code](https://github.com/MaciejMakowskiBille/NetflixClone/commit/1e9d40099cb4e4b904eaf00df4e823c9feb9466d)

File: Client/src/pages/signUp/components/payments/_payments-all.scss (hunk 8 above)
Comment (StachuDev) [reply Client/src/pages/signUp/components/payments/_payments-all.scss:26]: [add style colors as sass variables, little html tag fixes](https://github.com/MaciejMakowskiBille/NetflixClone/commit/128c425d91777274c7029ff023305206ce612349)

File: Client/src/style/fonts/roboto/LICENSE.txt (hunk 9 above)
Comment (StachuDev) [reply Client/src/style/fonts/roboto/LICENSE.txt]: commit: [tidy up code, pageArray type name change](https://github.com/MaciejMakowskiBille/NetflixClone/commit/7f820804210f6dd4d5f4568d0d2fead3afae1a44)

File: Client/src/utils/Posts.ts (hunk 10 above)
Comment (StachuDev) [reply Client/src/utils/Posts.ts:41]: commit: [tidy up code, pageArray type name change](https://github.com/MaciejMakowskiBille/NetflixClone/commit/7f820804210f6dd4d5f4568d0d2fead3afae1a44)

File: Client/src/pages/signUp/components/payments/_payments-all.scss (hunk 8 above)
Comment (StachuDev) [reply Client/src/pages/signUp/components/payments/_payments-all.scss:26]: [little fade-in change](https://github.com/MaciejMakowskiBille/NetflixClone/pull/12/commits/e7b779b644ed1295072fefd055f86f4db2deceae)

**Review by MaksymAndriienko — APPROVED** [review]

**Pull request merged**